src/test/scala/com/fulcrumgenomics/fastq/DemuxFastqsTest.scala

```diff
@@ -964,6 +964,60 @@ class DemuxFastqsTest extends UnitSpec with OptionValues with ErrorLogLevel {
     recR2.apply[String]("BC") shouldBe "AAAAAAAA-GATTACAGA"
   }
 
+
+  def testEndToEndBarcodeMetrics(fastqStandards: FastqStandards, omitFailingReads: Boolean = false, omitControlReads: Boolean = false): Unit = {
+    // Build the FASTQ
+    val fastqs = new ListBuffer[FastqRecord]()
+    val namePrefix = "Instrument:RunID:FlowCellID:Lane:Tile:X"
+    val qualities = "?"*17 + Range.inclusive(2, 40).map(q => (q + 33).toChar).mkString
+    fastqs += fq(name=f"$namePrefix:1", comment=Some("1:Y:0:SampleNumber"), bases=sampleBarcode1 + "A"*39, quals=Some(qualities)) // matches the first sample -> first sample
+    fastqs += fq(name=f"$namePrefix:2", comment=Some("2:N:0:SampleNumber"), bases=sampleBarcode1 + "G"*39, quals = Some(qualities)) // matches the first sample -> first sample
+//    fastqs += fq(name=f"$namePrefix:2", comment=Some("2:Y:0:SampleNumber"), bases=sampleBarcode2 + "G"*39, quals = Some(qualities)) // matches the second sample -> second sample
+    val metricsFilename = "demux_barcode_metrics.txt"
```

Comment:
Please make a temporary file, and then specify it to `DemuxFastqs`
```suggestion
    val metricsFilename = makeTempFile("demux_barcode_metrics", ".txt")
```